create_laion_ds.py

```diff
@@ -0,0 +1,119 @@
+from ann_benchmarks.algorithms.bruteforce import BruteForceBLAS
+import numpy as np
+import click
+import h5py
+from joblib import Parallel, delayed
+import multiprocessing
+import tqdm
+from urllib.request import urlretrieve
+import sklearn.model_selection
+import os
+import wget
+import sys
+
+
+def download(src, dst):
+    if not os.path.exists(dst):
+        # TODO: should be atomic
+        print("downloading %s -> %s..." % (src, dst))
+        wget.download(src, dst)
+
+
+def calc_i(i, x, bf, test, neighbors, distances, count):
+    if i % 1000 == 0:
+        print("%d/%d..." % (i, len(test)))
+    res = list(bf.query_with_distances(x, count))
+    res.sort(key=lambda t: t[-1])
+    neighbors[i] = [j for j, _ in res]
+    distances[i] = [d for _, d in res]
+
+
+def calc(bf, test, neighbors, distances, count):
+    Parallel(n_jobs=multiprocessing.cpu_count(), require="sharedmem")(
```

Comment:
While creating hdf5 files I encountered an issue: 
```
train size:    100000 *  512
test size:      10000 *  512
0/10000...
BLAS : Program is Terminated. Because you tried to allocate too many memory regions.
BLAS : Program is Terminated. Because you tried to allocate too many memory regions.
BLAS : Program is Terminated. Because you tried to allocate too many memory regions.
…
Segmentation fault (core dumped)
```
The system that I was using is Ubuntu 22.04 with 144 cpu_count and python3.8.10. I was able to workaround this error with hardcoding the n_jobs to 64.